### ISSUE TITLE
Crash in locale code on startup

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#6870]
This is the current top actionable crash on play console, 113 in last 30 days

###### Reproduction Steps

1. start the app
2. database error?
3. crash

###### Expected Result

App runs

###### Actual Result

Stack traces from play console - I think it's pre-ACRA so this might be the only way to get them

```
java.lang.NullPointerException: 
  at com.ichi2.anki.AnkiDroidApp.getAppResources (AnkiDroidApp.java:330)
  at com.ichi2.anki.dialogs.AsyncDialogFragment.res (AsyncDialogFragment.java:22)
  at com.ichi2.anki.dialogs.DatabaseErrorDialog.getTitle (DatabaseErrorDialog.java:342)
  at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog (DatabaseErrorDialog.java:64)
  at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog (DatabaseErrorDialog.java:23)
  at androidx.fragment.app.DialogFragment.onGetLayoutInflater (DialogFragment.java:380)
  at androidx.fragment.app.Fragment.performGetLayoutInflater (Fragment.java:1412)
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:881)
  at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManagerImpl.java:1238)
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:1303)
  at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange (FragmentManagerImpl.java:2659)
  at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated (FragmentManagerImpl.java:2613)
  at androidx.fragment.app.FragmentController.dispatchActivityCreated (FragmentController.java:246)
  at androidx.fragment.app.FragmentActivity.onStart (FragmentActivity.java:542)
  at androidx.appcompat.app.AppCompatActivity.onStart (AppCompatActivity.java:210)
  at com.ichi2.anki.AnkiActivity.onStart (AnkiActivity.java:94)
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1392)
  at android.app.Activity.performStart (Activity.java:7260)
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:3008)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:180)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:165)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:142)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1839)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:201)
  at android.app.ActivityThread.main (ActivityThread.java:6864)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
```

Possibly related, for another 54 crashes over 30 days

```
java.lang.NullPointerException: 
  at android.app.ActivityThread.updateLocaleListFromAppContext (ActivityThread.java:6168)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6415)
  at android.app.ActivityThread.access$1300 (ActivityThread.java:222)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1862)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7458)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:935)
```

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison-1** [comment]
This implies that `AnkiDroidApp.sInstance` is null. I'm stumped.

**Comment by mikehardy** [comment]
I recall reading something about ContentProviders being installed prior to absolutely anything as a weird lifesycle quirk, and in fact some crash reporting frameworks (crashlytics at least) appear to use that to init prior to app init. No idea of that info is useful.

**Comment by david-allison-1** [comment]
Most likely! Thanks, I'll take a look

**Comment by mikehardy** [comment]
we should hopefully get better information in 2.13 release builds now that #6891 is in with earlier init

**Comment by david-allison-1** [comment]
Fixed in 04408c1329c349d68bc3b6d63fab760f9767a7d6 #7374

**Comment by david-allison-1** [comment]
To replicate:

Activities can be launched without `onCreate` when an app is being backed up:

https://stackoverflow.com/questions/57211036/why-backup-related-process-might-cause-applications-oncreate-is-not-executed
